lewis/examples/modbus_device/__init__.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# *********************************************************************
+# lewis - a library for creating hardware device simulators
+# Copyright (C) 2016 European Spallation Source ERIC
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# *********************************************************************
+
+from lewis.devices import Device
+from lewis.adapters.modbus import ModbusAdapter, ModbusDataBank
+
+
+class ModbusDevice(Device):
+    pass
+
+
+class ModbusInterface(ModbusAdapter):
+    """
+    The class attributes di, co, ir and hr represent Discrete Inputs, Coils, Input Registers and
+    Holding Registers, respectively. Each attribute should be assigned a ModbusDataBank instance
+    by the Interface implementation.
+
+    Here, two basic ModbusDataBanks are created and initialized to a default value across the full
+    range of valid addresses. One DataBank is shared by di and co, and the other by ir and hr to
+    demonstrate overlaid memory segments. If you want each segment to have its own memory, just
+    create separate instances for all four.
+    """
+    di = ModbusDataBank.create_basic(False)
+    co = di
+    ir = ModbusDataBank.create_basic(0)
+    hr = ir
```

Comment:
I have the feeling that we should be very, very careful with labeling this as a general example for a Modbus device given that there is basically no support for exposing actual device behavior yet. Correct me if I'm wrong, but this is basically the only possible device at the moment, right? The only reason it's required there is that we once said "A Simulation requires a device".

Could we maybe move this to `devices` and call it something like `modbus_echo` or something like that? Combined with the logging functionality it would be very useful for diagnosing IOC-level problems concerning correct addresses and transmitted values and so on - and it would never change.

But once there are binding tools like for the other protocols, this will change significantly, so from my point of view it would be better to not put this into the "examples" module.